app/services/course/assessment/submission/update_service.rb

```diff
@@ -9,6 +9,35 @@ def update
     end
   end
 
+  def delete_file
```

Comment:
The adding/deletion of files are programming answer specific. Should shift both these actions to answers controller, put it under `submission/answer/programming/xxx`. 

On a second thought, I think you don't need these two actions, there's http://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html from rails and we have already adopted it to answers, that's why we can update files in the answer.